src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj

```diff
@@ -0,0 +1,285 @@
+
+    <!-- Copy assets to output path from NuGet packages. -->
+    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
+
+    <!-- Workaround for the way Microsoft.AspNetCore.Server.IIS.csproj references ANCM native deps. Do not copy content from ProjectReference's CopyToOutput items.  -->
+    <_GetChildProjectCopyToOutputDirectoryItems>false</_GetChildProjectCopyToOutputDirectoryItems>
+
+    <!-- This project should not be referenced via the `<Reference>` impementation. -->
+    <IsProjectReferenceProvider>false</IsProjectReferenceProvider>
+
+    <!-- Properties related to crossgen -->
+    <CrossGenSymbolsType>CreatePerfMap</CrossGenSymbolsType>
+    <CrossGenSymbolsType Condition="'$(TargetOsName)' == 'win'">CreatePDB</CrossGenSymbolsType>
+
+    <LibPrefix Condition=" '$(TargetOsName)' != 'win' ">lib</LibPrefix>
+    <LibExtension>.so</LibExtension>
+    <LibExtension Condition=" '$(TargetOsName)' == 'win' ">.dll</LibExtension>
+    <LibExtension Condition=" '$(TargetOsName)' == 'osx' ">.dylib</LibExtension>
+    <ExeExtension Condition=" '$(TargetOsName)' == 'win' ">.exe</ExeExtension>
+    <!-- 3B = semicolon in ASCII -->
+    <PathSeparator Condition="'$(PathSeparator)' == ''">:</PathSeparator>
+    <PathSeparator Condition=" '$(TargetOsName)' == 'win' ">%3B</PathSeparator>
+
+    <!-- Crossgen executable name -->
+    <CrossgenToolFileName>crossgen</CrossgenToolFileName>
+    <CrossgenToolFileName Condition=" '$(TargetOsName)' == 'win' ">$(CrossgenToolFileName).exe</CrossgenToolFileName>
+    <!-- Default crossgen executable relative path -->
+    <CrossgenToolPackagePath>$(CrossgenToolFileName)</CrossgenToolPackagePath>
+    <!-- Disambiguated RID-specific crossgen executable relative path -->
+    <CrossgenToolPackagePath Condition=" '$(TargetRuntimeIdentifier)' == 'linux-arm' ">x64_arm\$(CrossgenToolPackagePath)</CrossgenToolPackagePath>
+    <CrossgenToolPackagePath Condition=" '$(TargetRuntimeIdentifier)' == 'linux-arm64' ">x64_arm64\$(CrossgenToolPackagePath)</CrossgenToolPackagePath>
+    <CrossgenToolPackagePath Condition=" '$(TargetRuntimeIdentifier)' == 'win-arm' ">x86_arm\$(CrossgenToolPackagePath)</CrossgenToolPackagePath>
+
+    <RuntimePackageRoot>$(NuGetPackageRoot)/runtime.$(RuntimeIdentifier).microsoft.netcore.app/$(MicrosoftNETCoreAppPackageVersion)/</RuntimePackageRoot>
+    <CrossgenToolPath>$(RuntimePackageRoot)tools/$(CrossgenToolPackagePath)</CrossgenToolPath>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Reference Include="@(Dependency);@(ExternalDependency)" />
+    <ProjectReference Condition="'$(BuildIisNativeProjects)' == 'true' AND '$(BuildNative)' != 'false'" Include="$(RepositoryRoot)src\Servers\IIS\AspNetCoreModuleV2\InProcessRequestHandler\InProcessRequestHandler.vcxproj">
+      <SetPlatform>Platform=$(TargetArchitecture)</SetPlatform>
+      <SetPlatform Condition="'$(TargetArchitecture)' == 'x86'">Platform=Win32</SetPlatform>
```

Comment:
Slightly unrelated: Might be good to define `TargetNativeArchitecture` and `NativePlatform` so we can use it in native project references without conditions.

Reply:
I think it should be done too, but it's not trivial. Let's use https://github.com/aspnet/AspNetCore/issues/7196 to follow up on this.

Reply:
I'll let you chose here. I'm just worried about supporting all these slight modifications of one thing in many places.

Reply:
Commend here was intended for https://github.com/aspnet/AspNetCore/pull/7475#discussion_r256009498 thread.

Reply:
I'd like to leave this as is. Once we have confirmed that this new package is sufficient to replace the `runtime.<rid>` versions, we probably don't need to have as many NativeProjectReference's. I'd like to move most of our tests on to using the shared framework instead of having app-local versions of ANCM.